tests/Unit/Evolution/DiscontinuousGalerkin/Test_ObserveErrorNorms.cpp

```diff
@@ -0,0 +1,337 @@
+    results.observation_id = observation_id;
+    results.subfile_name = subfile_name;
+    results.reduction_names = reduction_names;
+    results.time = std::get<0>(reduction_data.data());
+    results.number_of_grid_points = std::get<1>(reduction_data.data());
+    results.errors.clear();
+    tmpl::for_each<tmpl::range<size_t, 2, sizeof...(Ts)>>([&reduction_data](
+        const auto index_v) noexcept {
+      constexpr size_t index = tmpl::type_from<decltype(index_v)>::value;
+      results.errors.push_back(std::get<index>(reduction_data.data()));
+    });
+  }
+};
+
+MockContributeReductionData::Results MockContributeReductionData::results{};
+
+template <typename System>
+struct Metavariables;
+
+template <typename System>
+struct ElementComponent {
+  using component_being_mocked = void;
+
+  using metavariables = Metavariables<System>;
+  using chare_type = ActionTesting::MockArrayChare;
+  using array_index = int;
+  using const_global_cache_tag_list = tmpl::list<>;
+  using initial_databox = db::compute_databox_type<db::AddSimpleTags<>>;
+  using action_list = tmpl::list<>;
+};
+
+template <typename System>
+struct MockObserverComponent {
+  using component_being_mocked = observers::Observer<Metavariables<System>>;
+  using replace_these_simple_actions =
+      tmpl::list<observers::Actions::ContributeReductionData>;
+  using with_these_simple_actions = tmpl::list<MockContributeReductionData>;
+
+  using metavariables = Metavariables<System>;
+  using chare_type = ActionTesting::MockArrayChare;
+  using array_index = int;
+  using const_global_cache_tag_list = tmpl::list<>;
+  using initial_databox = db::compute_databox_type<db::AddSimpleTags<>>;
+  using action_list = tmpl::list<>;
+};
+
+template <typename System>
+struct Metavariables {
+  using system = System;
+  using component_list =
+      tmpl::list<ElementComponent<System>, MockObserverComponent<System>>;
+  using const_global_cache_tag_list = tmpl::list<
+      OptionTags::AnalyticSolution<typename System::solution_for_test>>;
+
+  struct ObservationType {};
+  using element_observation_type = ObservationType;
+};
+
+// Test systems
+
+struct ScalarSystem {
+  struct ScalarVar : db::SimpleTag {
+    static std::string name() noexcept { return "Scalar"; }
+    using type = Scalar<DataVector>;
+  };
+
+  static constexpr size_t volume_dim = 1;
+  using vars_for_test = tmpl::list<ScalarVar>;
+  struct solution_for_test {
+    template <typename CheckTensor>
+    static void check_data(const CheckTensor& check_tensor) noexcept {
+      check_tensor("ErrorScalar", ScalarVar{});
+    }
+
+    tuples::tagged_tuple_from_typelist<vars_for_test> variables(
+        const tnsr::I<DataVector, 1>& x, const double t,
+        const vars_for_test /*meta*/) const noexcept {
+      return {Scalar<DataVector>{1.0 - t * get<0>(x)}};
+    }
+
+    void pup(PUP::er& /*p*/) noexcept {}  // NOLINT
+  };
+};
+
+struct ComplicatedSystem {
```

Comment:
Since a lot of this code is shared with `ObserveFields`, do you think it makes sense to factor it into a helper? Keeping in mind what you have planned in coming PRs for these observe events

Reply:
I'll look into this.  I'm not sure off the top of my head.

Reply:
Did you get a chance to look at this, @wthrowe?

Reply:
I decided the answer really wasn't clear so I'd put it off until we have a better idea of what a more permanent form of the observation setup is going to be.

Reply:
Sounds like a good plan to me :+1: